packages/workspace/src/browser/quick-open-workspace.ts

```diff
@@ -28,11 +29,15 @@ export class QuickOpenWorkspace implements QuickOpenModel {
     @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService;
     @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService;
     @inject(MessageService) protected readonly messageService: MessageService;
+    @inject(FileSystem) protected readonly fileSystem: FileSystem;
+    protected readonly workspace: string | undefined;
 
-    open(workspaces: string[]): void {
+    async open(workspaces: string[]): Promise<void> {
         this.items = [];
+        const home = await this.fileSystem.getCurrentUserHome().then(stat => stat ? new URI(stat.uri).withoutScheme().toString() : undefined);
```

Comment:
If you use the `.then(...)` form, you also need a `.catch(...)`.  You have probably seen this message before:

```
(node:12964) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): 1
(node:12964) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
This means that if `this.fileSystem.getCurrentUserHome()` fails, we'll get this warning, and in future versions of node it will kill the process.  I would suggest doing it in two steps like this:

```
        const homeStat = await this.fileSystem.getCurrentUserHome();
        const home = homeStat ? new URI(homeStat.uri).withoutScheme().toString() : undefined;
 ```